wsmaster/che-core-api-devfile/src/main/resources/schema/devfile.json

```diff
@@ -361,6 +362,54 @@
                 }
               }
             }
+          },
+          "entrypoints": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "deployment": {
+                  "type": "string",
+                  "description": "The name of the deployment (either the concrete name or the generateName from the metadata). If not specified, then all deployments are considered."
+                },
+                "pod": {
+                  "type": "string",
+                  "description": "The name of the pod (either the concrete name or the generateName from the metadata) containing the container to apply the entrypoint to. Optional if there is just one pod in the selected deployment (or as a top level entry in the list)."
+                },
+                "container": {
+                  "type": "string",
+                  "description": "The name of the container to apply the entrypoint to. Optional if there is just a single container in the selected pods."
+                },
+                "deploymentSelector": {
```

Comment:
Looks like `deploymentSelector` and `podSelector` are not listed in https://github.com/eclipse/che/issues/12693
Could you please add the corresponding comment in the issue(or even change issue description) and agree on the format with @l0rd.

Question: what if this selector contradict selector of tool and tool won't contain needed objects? Then Che Server will throw an exception or just do nothing?

Reply:
> what if this selector contradict selector of tool and tool won't contain needed objects? Then Che Server will throw an exception or just do nothing?

It'll just do nothing.

Reply:
I think it's better to throw an exception, otherwise, it might be an exercise for a user to understand why workspace does not work in the way he wants in case of error like a simple typo.

Reply:
> Could you please add the corresponding comment in the issue(or even change issue description) and agree on the format with @l0rd.

https://github.com/eclipse/che/issues/12693#issuecomment-471514225

Reply:
> I think it's better to throw an exception, otherwise, it might be an exercise for a user to understand why workspace does not work in the way he wants in case of error like a simple typo.

That's a fair point. I'll make the changes.

Reply:
Hmm.. I'd like to make this part of devfile validation yet during that we don't know the content of the referenced files.

This is gonna require more work to do right.